packages/http-client/src/middleware/timeout.ts

```diff
@@ -0,0 +1,49 @@
+import {
+  Context,
+  Middleware,
+  MiddlewareIn,
+  MiddlewareOut
+} from './middleware-type';
+
+const defaultTimeout = 10000;
+
+/**
+ * Helper method to build a timout middleware.
+ * @param timeoutValue - Timeout in milliseconds. Default value are 10 seconds.
+ * @returns The middleware adding a timeout to the function.
+ */
+export function timeout<ReturnType, ContextType extends Context>(
```

Comment:
[req] The implementation of the default middleware should be a separate package so when dependencies are required (curcuit breaker, retries, ...) a user who does not use the default does not need to install those deps.

The middleware param types are a property of http client in my opinion as the `http-client` defines the spec of what it accepts and the `default-middleware` would only implement against that spec.